packages/blockchain-wallet-v4-frontend/src/modals/Swap/EnterAmount/index.tsx

```diff
@@ -128,14 +127,24 @@ class EnterAmount extends PureComponent<Props> {
                   <Option
                     role='button'
                     data-e2e='selectFromAcct'
-                    onClick={() =>
-                      this.props.swapActions.setStep({
-                        step: 'COIN_SELECTION',
-                        options: {
-                          side: 'BASE'
-                        }
-                      })
-                    }
+                    onClick={() => {
+                      const isAccountZeroBalance = checkAccountZeroBalance(
+                        val.accounts
+                      )
+
+                      if (isAccountZeroBalance) {
+                        this.props.swapActions.setStep({
+                          step: 'NO_HOLDINGS'
+                        })
+                      } else {
+                        this.props.swapActions.setStep({
+                          step: 'COIN_SELECTION',
+                          options: {
+                            side: 'BASE'
+                          }
+                        })
+                      }
+                    }}
```

Comment:
might be good to move all stuff into separate function

Reply:
Fixed!